Framework/Core/include/Framework/TableTreeHelpers.h

```diff
@@ -22,97 +22,205 @@ namespace o2::framework
 {
 
 // -----------------------------------------------------------------------------
-// TableToTree allows to save the contents of a given arrow::Table to a TTree
-//  BranchIterator is used by TableToTree
+// TableToTree allows to save the contents of a given arrow::Table into
+// a TTree
+// ColumnToBranch is used by GenericTableToTree
 //
 // To write the contents of a table ta to a tree tr on file f do:
-//  . TableToTree t2t(ta,f,treename);
-//  . t2t.addBranch(coumn1); t2t.addBranch(coumn1); ...
-//    OR
-//    t2t.addAllBranches();
+//  . GenericTableToTree t2t(ta, f,treename);
+//  . t2t.addBranches();
+//    OR t2t.addBranch(column.get(), field.get()), ...;
 //  . t2t.process();
 //
 // .............................................................................
-class BranchIterator
+class ColumnToBranchBase
 {
+ public:
+  ColumnToBranchBase(arrow::ChunkedArray* column, arrow::Field* field, int size = 1);
+  virtual ~ColumnToBranchBase() = default;
+  void at(int64_t* pos)
+  {
+    mCurrentPos = pos;
+    resetBuffer();
+  }
 
- private:
-  std::string mBranchName;    // branch name
-  arrow::ArrayVector mChunks; // chunks
-  Int_t mNumberChuncs;        // number of chunks
-  Int_t mCounterChunk;        // chunk counter
-  Int_t mNumberRows;          // number of rows
-  Int_t mCounterRow;          // row counter
-
-  // data buffers for each data type
-  bool mStatus = false;
-  arrow::Field* mField;
-  arrow::Type::type mFieldType;
-  arrow::Type::type mElementType;
-  int32_t mNumberElements;
-  std::string mLeaflistString;
-
-  TBranch* mBranchPtr = nullptr;
-
-  char* mBranchBuffer = nullptr;
-  void* mValueBuffer = nullptr;
-
-  std::shared_ptr<arrow::BooleanArray> mArray_o = nullptr;
-  //bool mBoolValueHolder;
-  bool* mVariable_o = nullptr;
-
-  uint8_t* mVariable_ub = nullptr;
-  uint16_t* mVariable_us = nullptr;
-  uint32_t* mVariable_ui = nullptr;
-  uint64_t* mVariable_ul = nullptr;
-  int8_t* mVariable_b = nullptr;
-  int16_t* mVariable_s = nullptr;
-  int32_t* mVariable_i = nullptr;
-  int64_t* mVariable_l = nullptr;
-  float* mVariable_f = nullptr;
-  double* mVariable_d = nullptr;
-
-  // initialize a branch
-  bool initBranch(TTree* tree);
-
-  // initialize chunk ib
-  bool initDataBuffer(Int_t ib);
+ protected:
+  std::string mBranchName;
```

Comment:
Please move this to a separate file.

Reply:
You mean move `CoumnToBranchBase`/`ColumnToBranch` classes?

Reply:
Moved to .cxx